### ISSUE TITLE
RenderSettings.defaultTimeZone is unused

### ISSUE THREAD

**Issue by mbaumbach** [bkiers/Liqp#218]
It looks like this may have been introduced to allow setting the timezone (similar to the locale) for the `Date` filter, but the `Date` filter is only using the `ZoneId.systemDefault()` when it comes across a time that is in seconds. Is this meant to be used in that case or an unused property on the `RenderSettings`? It was added in commit a668168e35dfc3938824a96c5feb9ebf4116799d.

Liqp version: 0.8.0

**Comment by mbaumbach** [comment]
Perhaps @msangel knows the intention here. I've written a custom time zone aware date filter for our own code base, but it would be awesome to make use of this directly from the library!

**Comment by msangel** [comment]
Agree. My mistake. You are right on the purpose of this field but looks like I forget to actually use it. 
Will do a fix in a few days.
Thankls!

**Comment by msangel** [comment]
tl;dr: it's big and breaking change

@mbaumbach 
The issue seems more complicated than expected.
![Screenshot_2021-08-20_04-10-53](https://user-images.githubusercontent.com/773163/130163422-eb996824-b111-4d03-8ddd-150b58b68453.png)
since the default timezone is also used in https://github.com/bkiers/Liqp/blob/master/src/main/java/liqp/filters/date/Parser.java#L99
and it's logically to use existed parametrized timezone
but
that function is used everywhere because we convert an arbitrary value to proper time if it is required so, following the same ruby rule: if some date part is not set - the current(and in our case "default") should be used.

most used example:
https://github.com/bkiers/Liqp/blob/master/src/main/java/liqp/LValue.java#L263

and it's true that even in that case the templator should use defined "default time zone", but then the `toString` function must accept it, and....
![Screenshot_2021-08-20_04-10-21](https://user-images.githubusercontent.com/773163/130164197-bfe6fc7a-cf49-4775-8ac7-46e1811c751e.png)

**Comment by msangel** [comment]
Made a pull request for that actually: https://github.com/bkiers/Liqp/pull/219
but lets the @bkiers  decide if it is worse to break backward compatibility

Minimal fix for an exactly described case already in master: https://github.com/bkiers/Liqp/commit/d35be5aea36cddb55925b09d6289afbf223fb855

**Comment by msangel** [comment]
I would say: if people need a not-default timezone, lets them use datatypes with explicit timezone. But maybe I just don't want to break compatibility. But since this is a big jump anyway, it may be worth it.

**Comment by mbaumbach** [comment]
@msangel If you construct the parser without specifying a timezone and the parser continues to default to using the system default, is it still a breaking change? Wouldn't this require people to opt-in by providing a TimeZone at the time of constructing the parser? Since that wasn't an available API before, it sounds like it might not break compatibility, but maybe I'm missing something.

Thanks for working on this!

**Comment by msangel** [comment]
What you're saying is correct. External interfaces aren`t changed even a bit. 
But the internal ones do. So if someone uses their own filters or tags it's a big chance of them being broken.

**Comment by mbaumbach** [comment]
Ah, that makes sense. Thanks for explaining the use case!

**Comment by msangel** [comment]
Test case with explaining what's the fix do, this works on [proposed code](https://github.com/bkiers/Liqp/pull/219), but fails on master:
```java
@Test
public void testAppendToDateTypeWithDefaultTimezoneSet() {
    LocalDateTime time = LocalDateTime.of(2020,1,1, 12,59,59, 999);
    Map<String, Object> data = Collections.singletonMap("a", time);

    // -5 here (uses fixed offset as don't want to play with DST)
    ZoneId defaultTimeZone = ZoneId.ofOffset("UTC", ZoneOffset.ofHours(-5));
        
    RenderSettings renderSettings = new RenderSettings.Builder().withDefaultTimeZone(defaultTimeZone).build();
        
    // is -0500 here
    String res = Template.parse("{{ '!' | append: a }}").withRenderSettings(renderSettings).render(data);
    assertEquals("!2020-01-01 12:59:59 -0500", res);

    // is -0500 here
    res = Template.parse("{{ a | append: '!' }}").withRenderSettings(renderSettings).render(data);
    assertEquals("2020-01-01 12:59:59 -0500!", res);
}
```
on mater it fails with this exception(as I live in +2)
![Screenshot_2021-08-20_17-28-52](https://user-images.githubusercontent.com/773163/130248841-03d979d2-deb9-45bb-80b9-30b4da4d9c8b.png)

In case of approval, I will cover all cases with tests(I like to write tests but on my paid work I almost never do this 😃 )

for @bkiers

**Comment by bkiers** [comment]
Thanks @msangel, nice work! Go for it :+1:
